internal/controller/deduper/deduper.go

```diff
@@ -156,24 +170,27 @@ func (d *Deduper) markRunning(id uuid.UUID) {
 	d.logger.Debug(
 		"markRunning: added previously unknown in-flight job",
 		zap.String("uuid", id.String()),
+		zap.String("source", source),
 		zap.Int("num-in-flight", numInFlight),
 	)
 }
 
-// markComplete records a job as not in-flight.
-func (d *Deduper) markComplete(id uuid.UUID) {
+// markNotRunning records a job as not in-flight.
+func (d *Deduper) markNotRunning(id uuid.UUID, source string) {
 	// Change state from in-flight to not in-flight.
 	numInFlight, ok := d.casa(id, false)
 	if !ok {
-		d.logger.Debug("markComplete: job was already not-in-flight!",
+		d.logger.Debug("markNotRunning: job was already not-in-flight!",
```

Comment:
Is this double negative something which is easy to understand?

Reply:
Fair point, I've renamed it back to `markFinished` (`markComplete` sounded to me as though the job succeeded, when it might not have).

Reply:
Or maybe it should be `unmarkRunning`? 🤔

Reply:
Maybe including the method name is not useful, given that zap includes the file and line number.